include/circt/Dialect/SV/SVInOutOps.td

```diff
@@ -48,6 +48,12 @@ def WireOp : SVOp<"wire", [DeclareOpInterfaceMethods<OpAsmOpInterface>]> {
 
   let assemblyFormat = "custom<ImplicitSSAName>(attr-dict) `:` type($result)";
   let hasCanonicalizer = true;
+
+  let extraClassDeclaration = [{
+    Type getElementType() {
+      return result().getType().cast<InOutType>().getElementType();
```

Comment:
I think that getType() should be typed as returning an inout type correctly, so this can just be:

`return getType().getElementType()`

Also the one for RegOp

Reply:
It is not currently.  Looks like it is just a constraint on the type and doesn't result in the storage type changing.